src/main/kotlin/CheckersModel.kt

```diff
@@ -0,0 +1,113 @@
+import kotlin.math.abs
+
+class CheckersModel() : BaseModel(8) {
+    private var whoMoves = Color.WHITE
+
+    override fun canMove(turn: BaseTurn): Pair<Boolean, Any?> {
+        if (turn.playerColor != whoMoves) {
+            //ходит не тот, кто должен бы
+            return Pair(false, null)
+        }
+        //TODO("Проверка выхода за границы")
+        val squareFrom = board[turn.from.first][turn.from.second]
+        val squareTo = board[turn.to.first][turn.to.second]
+        if (!squareFrom.color || !squareTo.color) {
+            return Pair(false, null)
+        }
+        if (squareFrom.figure != null) {
+            if (squareTo.figure != null) {
+                //клетка "куда" не пустая
+                return Pair(false, null)
+            }
+            if (squareFrom.figure!!.color != turn.playerColor) {
+                //фигура на клетку "откуда" не совпадает с цветом ходящего
+                return Pair(false, null)
+            }
+            val verticals = turn.to.second - turn.from.second
+            val horizontals = turn.to.first - turn.from.first
+            if ((horizontals == if (turn.playerColor == Color.WHITE) 1 else -1) && (abs(verticals) == 1)) {
+                return Pair(true, null)
+            }
+            if ((abs(horizontals) == 2) && (abs(verticals) == 2)
+                    && (board[turn.from.first][turn.from.second].figure!!.type == FigureType.Ordinary)) { //Если просто съедаем через клетку
+                val squareToEat = board[(turn.from.first + turn.to.first) / 2][(turn.from.second + turn.to.second) / 2]
+                if ((squareToEat.figure != null) && (squareToEat.figure?.color != squareFrom.figure!!.color)) { //проверяем что там стоит шашка другого цвета
+                    return Pair(true, squareToEat)
+                }
+            }
+            /*if ((abs(horizontals) == abs(verticals)) && (squareFrom.figure!!.type == FigureType.Queen)) { //съедаем через несколько клеток(должна быть дамкой)
+                var count: Int = 0
+                for (i in 1 until abs(horizontals) + 1) {
+
+                }
+            }*/
+            return Pair(false, null)
+        } else {
+            //клетка "откуда" пустая
+            return Pair(false, null)
+        }
+    }
+
+    override fun move(turn: BaseTurn) {
+        val canMoveResult = canMove(turn)
```

Comment:
If your method returns pair (or another data class), you may unpack it, as it was shown before.